visualization/app/codeCharta/ui/dialog/dialog.addScenarioSettings.component.ts

```diff
@@ -0,0 +1,107 @@
+import "./dialog.component.scss"
+import { AppSettings, DynamicSettings } from "../../codeCharta.model"
+import { StoreService } from "../../state/store.service"
+import { Scenario, ScenarioHelper } from "../../util/scenarioHelper"
+import { DialogService } from "./dialog.service"
+
+export interface AddAttributeContent {
+	metricName: string
+	currentMetric: string
+	isSelected: boolean
+	isDisabled: boolean
+	metricAttributeValue: any
+}
+
+export enum ScenarioCheckboxNames {
+	CAMERA_POSITION = "CameraPosition",
+	EDGE_METRIC = "Edge",
+	AREA_METRIC = "Area",
+	HEIGHT_METRIC = "Height",
+	COLOR_METRIC = "Color"
+}
+
+export class DialogAddScenarioSettingsComponent {
+	private _viewModel: {
+		scenarioName: string
+		currentAttribute: string
+		fileContent: AddAttributeContent[]
+	} = {
+		scenarioName: null,
+		currentAttribute: null,
+		fileContent: []
+	}
+
+	constructor(private $mdDialog, private storeService: StoreService, private dialogService: DialogService) {
+		this.initDialogFields()
+	}
+
+	public hide() {
+		this.$mdDialog.hide()
+	}
+
+	public addScenario() {
+		if (
+			ScenarioHelper.isScenarioExisting(this._viewModel.scenarioName) ||
+			this.isScenarioNameEmpty() ||
+			!this.isFileContentSelected()
+		) {
+			this.dialogService.showErrorDialog(
+				"Please select a non existing Scenario Name and select at least one attribute to be saved in the Scenario."
+			)
+		} else {
+			const selectedScenarioAttributes: AddAttributeContent[] = this._viewModel.fileContent.filter(x => x.isSelected == true)
+			const newScenario: Scenario = ScenarioHelper.createNewScenario(this._viewModel.scenarioName, selectedScenarioAttributes)
+			ScenarioHelper.addScenario(newScenario)
+			this.hide()
+		}
+	}
```

Comment:
I suggest to disable the `Add`-button as long as the scenarioName is empty or no scenarioCOntent is selected instead of the error pop-ups.